library/Core/src/Juvix/Core/Erased/Base/Types.hs

```diff
@@ -24,21 +30,28 @@ extensible
       | CatCoproductIntroRight (Term primVal)
       | CatCoproductElim (Term primVal) (Term primVal) (Term primVal) (Term primVal) (Term primVal)
       | Unit
+      | Record [ValField primVal]
       | Let NameSymbol.T (Term primVal) (Term primVal)
       | App (Term primVal) (Term primVal)
+      | RecElim [Symbol] (Term primVal) (Term primVal)
       deriving (Show, Eq, Generic)
 
+    type ValField primVal = Core.ValField' (Term primVal)
+
     data Type primTy
       = SymT NameSymbol.T
-      | Star Universe
+      | Star Core.Universe
       | PrimTy primTy
       | -- TODO: How to deal with dependency?
         Pi Usage.T (Type primTy) (Type primTy)
       | UnitTy
       | Sig Usage.T (Type primTy) (Type primTy)
       | CatProduct (Type primTy) (Type primTy)
       | CatCoproduct (Type primTy) (Type primTy)
+      | RecordTy [TypeField primTy]
```

Comment:
These new records are basically a blend of n-ary products and the □ of stuff like grtt, which is the basic type constructor for packing up multiple usages into one value. Our Σ basically has a □, but only on the left, which is weird and arbitrary but otherwise fine. It's just because I did it in analogy with Π without thinking about it too hard haha

A more principled decomposition would have been Σ with no usages and separate □, of course, and field names probably just not existing in core and later. But then having full records all the way down ended up being nicer in some ways.

I think Σ and Unit are on the way out since they're just special cases of records, which is why I never brought up changing it.